crypto/mem.c

```diff
@@ -165,6 +169,42 @@ void *OPENSSL_malloc(size_t size) {
   return ((uint8_t *)ptr) + OPENSSL_MALLOC_PREFIX;
 }
 
+void *OPENSSL_malloc_align_internal(size_t size, size_t alignment) {
+
+  // Only support up to 128-byte alignment. Current limit imposed by the type
+  // |align_offset_t|. |align_pointer| further requires |alignment| to be a
+  // power of 2.
+  if ((alignment == 0) ||
+      (alignment > 128) ||
+      ((alignment & (alignment - 1)) != 0)) {
+    return NULL;
+  }
+
+  // Memory is organised as follows
+  // +--------+----------------+-------------+---------------+
+  // | poison | offset padding | offset size | memory region |
+  // +--------+----------------+-------------+---------------+
+  // poison: added by |OPENSSL_malloc|.
+  // offset padding: offset bytes to align memory at |alignment| boundary.
+  // offset size: number of offset bytes needed, plus |ALIGN_OFFSET_SIZE|.
```

Comment:
If I understood correctly, then I'd suggest:
```suggestion
  // offset size: number of bytes in offset padding. This location is of size |ALIGN_OFFSET_SIZE|.
```